src/OrchardCore.Modules/OrchardCore.Security/Drivers/SecuritySettingsDisplayDriver.cs

```diff
@@ -0,0 +1,121 @@
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using OrchardCore.DisplayManagement.Entities;
+using OrchardCore.DisplayManagement.Handlers;
+using OrchardCore.DisplayManagement.Views;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Security.Options;
+using OrchardCore.Security.ViewModels;
+using OrchardCore.Settings;
+
+namespace OrchardCore.Security.Drivers
+{
+    public class SecuritySettingsDisplayDriver : SectionDisplayDriver<ISite, SecuritySettings>
+    {
+        internal const string SettingsGroupId = "SecurityHeaders";
+
+        private readonly IShellHost _shellHost;
+        private readonly ShellSettings _shellSettings;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly IAuthorizationService _authorizationService;
+
+        public SecuritySettingsDisplayDriver(
+            IShellHost shellHost,
+            ShellSettings shellSettings,
+            IHttpContextAccessor httpContextAccessor,
+            IAuthorizationService authorizationService)
+        {
+            _shellHost = shellHost;
+            _shellSettings = shellSettings;
+            _httpContextAccessor = httpContextAccessor;
+            _authorizationService = authorizationService;
+        }
+
+        public override async Task<IDisplayResult> EditAsync(SecuritySettings settings, BuildEditorContext context)
+        {
+            var user = _httpContextAccessor.HttpContext?.User;
+
+            if (!await _authorizationService.AuthorizeAsync(user, SecurityPermissions.ManageSecurityHeadersSettings))
+            {
+                return null;
+            }
+
+            return Initialize<SecuritySettingsViewModel>("SecurityHeadersSettings_Edit", model =>
+            {
+                model.ContentSecurityPolicy = settings.ContentSecurityPolicy;
```

Comment:
@Piedone if `ConfigureSecuritySettings()` is called I'm able now to bind settings in the `OrchardCoreBuilderExtensions` but I'm access the settings stored in the database all time. Any idea?

Reply:
Sorry, I don't understand the question.

Reply:
Seems we need to use `IOptions<SecuritySettings>` instead of `SecuritySettings` in the driver

Reply:
Yes, or a snapshot. Check out https://github.com/Lombiq/Orchard-Training-Demo-Module for a sample on how to do such options together with site settings.

Reply:
I already push a commit with the change before I look into your link, could you please review the PR it's ready from my side, I will try a little doc about the module